src/Dialogs/DataField.cpp

```diff
@@ -58,11 +60,13 @@ EditDataFieldDialog(const TCHAR *caption, DataField &df,
 
     gdf.ModifyValue(value);
     return true;
-  } else if (df.GetType() == DataField::Type::NUMBER) {
-    unsigned int value = df.GetAsInteger();
-    if (!NumberEntryDialog(caption, value, 5))  // 5 columns
-        return false;
-    df.SetAsInteger(value);
+  } else if (df.GetType() == DataField::Type::DATE) {
+    DataFieldDate &dfd = reinterpret_cast<DataFieldDate &>(df);
```

Comment:
This `reinterpret_cast` may crash XCSoar.

Reply:
A ```static_cast``` is much more exact! If this ok?

Reply:
This is not about being "exact". This is about being **correct**. And `reinterpret_cast` is not.
`static_cast` is correct, but more exact than what? What do you mean?